pkg/recipes/terraform/config/config.go

```diff
@@ -52,6 +60,84 @@ func New(moduleName string, envRecipe *recipes.EnvironmentDefinition, resourceRe
 	}
 }
 
+// getModuleSource is called to get the module source path with credential information
+// if template path represent a private source else returns the provided template path
+func getModuleSource(ctx context.Context, envConfig *recipes.Configuration, templatePath string, armOptions *arm.ClientOptions) (string, error) {
+	var source string
+	var err error
+	if strings.HasPrefix(templatePath, "git::") {
+		source, err = getGitSource(ctx, envConfig, templatePath, armOptions)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		source = templatePath
+	}
+	return source, err
+}
+
+// getGitSource creates a terraform module source in a generic git format with credential information
+// e.g: git::https://<username>:<pat>@<example.com>/<repo_path>
+func getGitSource(ctx context.Context, envConfig *recipes.Configuration, templatePath string, armOptions *arm.ClientOptions) (string, error) {
```

Comment:
Are we only supporting private registries that are accessed over HTTPS? Terraform supports other Git URLS (just looking at this: https://developer.hashicorp.com/terraform/language/modules/sources#github) so I'm wondering what the expected behavior would be if a user passed one of these formats